state/mongodb/mongodb.go

```diff
@@ -197,7 +208,12 @@ func (m *MongoDB) Delete(req *state.DeleteRequest) error {
 }
 
 func (m *MongoDB) deleteInternal(ctx context.Context, req *state.DeleteRequest) error {
-	filter := bson.M{id: req.Key}
+	var filter primitive.M
+	if req.ETag == nil {
+		filter = bson.M{id: req.Key}
+	} else {
+		filter = bson.M{id: req.Key, etag: req.ETag}
```

Comment:
Should `etag: req.ETag` be `etag: *req.ETag` to avoid and reflection logic the mongo driver may or may not have?

Reply:
I missed it was a pointer here. Fixed.